### ISSUE TITLE
Media Library 'featured image' dialogue missing link text that describes 'opens in new tab'.

### ISSUE THREAD

**Issue by anthonyburchell** [WordPress/gutenberg#17642]
**Describe the bug**
This was originally reported in https://core.trac.wordpress.org/ticket/48028

This github issue is being created because the bug was introduced via Gutenberg and will need to happen here to resolve the overall Accessibility bug. 

Description from Trac Report: 

> The "Media Library" panel of the "Featured Image" modal dialog contains a link labelled with "Edit image". This link takes users to a new window/tab, but that link does not inform users that it opens a new window or tab.
> A link does not naturally convey itself to assistive technology that it opens in a new tab without link text that describes it.
> It is important to convey that a link opens in a new tab because of the change of context may not be clear to everyone.
> Suggested solution
> Write more descriptive link text that describes the link opening in a new tab;
> Utilise the class of 'screen-reader-text' in CSS to hide that text (you can wrap the text in <span class="screen-reader-text">).
> Example of solution
> <a class="edit-attachment" href="..." target="_blank">
>   Edit Image
>   <span class="screen-reader-text">(Opens in a new window)</span>
> </a>
> 

This problem was originally raised on ticket [#47137](https://core.trac.wordpress.org/ticket/47137) and has been separated out here to reduce complexity.

**To reproduce**
1. Add an image block
2. Click on 'media library'
3. Select image
4. Click "Edit Image" in the sidebar
After 4 you will see this opens in a new tab. This is due to the fact that the required edit state logic is not currently in the MediaUpload component. 

**Expected behavior**
Instead of opening in a new tab, the link should just open the Edit Image state _within_ the same modal.

**Screenshots**
<img width="639" alt="Screen Shot 2019-09-29 at 4 35 55 PM" src="https://user-images.githubusercontent.com/11888980/65844836-632ce280-e2fd-11e9-8f14-06b5930958ef.png">

**Comment by anthonyburchell** [comment]
PR to resolve this is here: https://github.com/WordPress/gutenberg/pull/17641

**Comment by anthonyburchell** [comment]
Additionally, the PR above addresses another issue that needed to be fixed in order for the fix to work. Currently we use `return wp.media.view.MediaFrame.Post.extend( {` for the gallery views of the media library. We should instead be using 	`return wp.media.view.MediaFrame.Select.extend( {`
